### DISCUSSION TITLE
Sort by upload date

### DISCUSSION THREAD

**Discussion by enboig** [immich-app/immich#7559] [General]
I am uploading all my photos to immich, and inserting them into albums. Some of them have wrong date (or not date at all). Being able to sort them by upload date would help me to select them and add to an album.

Is there a way to sort them this way?

**Comment by AFumi39** [comment]
Have you found a way? I've the same problem, and I don't find any way to do this

**Comment by tobse91** [comment]
Cant find something like this too. When I import a large chunk of files, i want to check the ones i imported recently and a "sort by upload date" feature on the photos page would be very helpful.

**Comment by glowtape** [comment]
Ditto. I'm currently evaluating Immich, and the inability to sort by upload date makes tagging and organizing things into albums pretty annoying.

**Comment by thevrownguy** [comment]
I have been using Immich for a few months and just noticed I cannot do this. I had some pictures that were dated wrong and it took me AGES to find them in 60,000 customer photos.

**Comment by hirenshah** [comment]
I uploaded a load of photos to an album earlier and realised later on that ~50 of them were meant to be added to a different album.

Have spent ages trying to find the photos in question. Thought it would be fairly easy, but I was so wrong.

Hopefully the database has dates or something which I can use to get a list of all the uploads around a certain time. List + Powershell Script to find the files and then find them in Immich.

**Comment by superboo07** [comment]
I would love a feature like this please

**Comment by skatsubo** [comment]
Related FR https://github.com/immich-app/immich/discussions/8128: [Feature] Sort by Modified/Uploaded date in addition to the metadata defined date(created)

**Comment by SimJoSt** [comment]
This would also be great to have exposed as a search option.

**Comment by tobse91** [comment]
Finally it's coming, check out the [3.0.0 RC](https://pr-558.dev.immich.app/blog/v3.0.0-release)
